halo/genutil/evm/evm.go

```diff
@@ -184,15 +181,26 @@ func omegaPrefundAlloc() types.GenesisAlloc {
 // from the native bridge's prefund balance, and transferring the same amount
 // to the bridge contract on L1.
 func mainnetPrefundAllocs() types.GenesisAlloc {
-	return types.GenesisAlloc{
-		eoa.MustAddress(netconf.Mainnet, eoa.RoleCreate3Deployer): {Balance: eth0_1}, // deploys one contract
-		eoa.MustAddress(netconf.Mainnet, eoa.RoleDeployer):        {Balance: eth0_1}, // deploys a couple contracts
-		eoa.MustAddress(netconf.Mainnet, eoa.RoleManager):         {Balance: eth10},  // rarely used
-		eoa.MustAddress(netconf.Mainnet, eoa.RoleUpgrader):        {Balance: eth10},  // rarely used
-		eoa.MustAddress(netconf.Mainnet, eoa.RoleRelayer):         {Balance: eth100}, // enough to relay for a while
-		eoa.MustAddress(netconf.Mainnet, eoa.RoleMonitor):         {Balance: eth100}, // enough to monitor for a while
-		eoa.MustAddress(netconf.Mainnet, eoa.RoleFunder):          {Balance: eth500}, // enough for a few funding actions
+	allocs := make(types.GenesisAlloc)
+
+	fundRoles := []eoa.Role{
+		eoa.RoleRelayer, eoa.RoleMonitor,
+		eoa.RoleCreate3Deployer, eoa.RoleManager,
+		eoa.RoleUpgrader, eoa.RoleDeployer,
+		eoa.RoleFunder,
 	}
+
+	for _, role := range fundRoles {
+		fund, ok := eoa.GetFundThresholds(tokens.OMNI, netconf.Mainnet, role)
+		if !ok {
```

Comment:
This will skip the `eoa.Funder` role, because we do not set thresholds for it. Either choose a sensible value, or derive it from the target balances of other roles